include/kernel/thread.h

```diff
@@ -145,9 +150,11 @@ static inline void commitTime(void)
         }
         NODE_STATE(ksCurSC)->scConsumed += NODE_STATE(ksConsumed);
     }
-    if (CONFIG_NUM_DOMAINS > 1) {
+    /* If we have switched domains, ksDomainTime has already been reset and is accounted for the
+     * old domain. If we have not switched domains, we still need to account for the time that
+     * has passed. */
+    if (CONFIG_NUM_DOMAINS > 1 && !switchedDomain) {
         assert(ksDomainTime > NODE_STATE(ksConsumed));
```

Comment:
This assert needs to reflect the condition of `isCurDomainExpitred`. It would be easiest to just use that function for the assert.

Reply:
Ah, I see. This should indeed be true as `ksDomainTime` here should be greater than `NODE_STATE(ksConsumed) + getTimerPrecision()`. In which case the condition for the following block should always be false.

Reply:
Yes, agreed. That's the bit we'd want to remove when the proof has shown that our collective arguments for that are correct.